File: C++/level2/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> work_day;
    int index;

    while(!progresses.empty()){
        if((100-progresses.back())%speeds.back() == 0)
            work_day.push_back((100-progresses.back())/speeds.back());
        else
            work_day.push_back((100-progresses.back())/speeds.back()+1);
        progresses.pop_back();
        speeds.pop_back();
    }

    while(!work_day.empty()){
        int temp = work_day.back();
        work_day.pop_back();
        index = 1;
        for(int i=work_day.size()-1; i>=0; i--){
            if(temp >= work_day[i]){
                work_day.erase(work_day.begin()+i);
                index++;
            }
            else
                break;
        }
        answer.push_back(index);
    }
    
    return answer;
}

File: JavaScript/level2/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let current = 0;
    let cache = [];
    let index=0;
    
    for(let i=0; i<progresses.length; i++)
        cache.push(Math.ceil((100 - progresses[i])/speeds[i]));
    
    for(let i of cache){
        if(current < i){
            if(index != 0)
                answer.push(index);
            current = i;
            index = 1;
        }
        else 
            index++;
    }
    
    answer.push(index);

    return answer;
}